arcade_machine/input_manager.py

```diff
@@ -0,0 +1,212 @@
+
+# Input manager to interpret inputs from keyboard, or joystick(s) and cast them to standard keys for all games to use.
+"""
+Arcade console to have (1) joystick with 4-directional control and (4) buttons per player.
+Each connected joystick will have an instance_id, for this console it is 0, or 1 for (2) players.
+The interpretations of the buttons from the keyboard and the joystick(s) are listed:
+
+Button  |  Keyboard  |  Joystick        |  Output
+        |  P1, P2    |  P1 (0), P2 (1)  |  Press & Release
+-----------------------------------------------------
+        |            | AXIS MOTION      |
+UP      | K_w, K_i   | 0:1, 1:1         | PX_N_DOWN & PX_NS_UP
+DOWN    | K_s, K_k   | 0:1, 1:1         | PX_S_DOWN & PX_NS_UP
+LEFT    | K_a, K_j   | 0:0, 1:0         | PX_W_DOWN & PX_EW_UP
+RIGHT   | K_d, K_l   | 0:0, 1:0         | PX_E_DOWN & PX_EW_UP
+        |            | STANDARD BUTTON  |
+A       | K_1, K_7   | 0:0, 1:6         | PX_A_DOWN & PX_A_UP
+B       | K_2, K_8   | 0:1, 1:7         | PX_B_DOWN & PX_B_UP
+X       | K_3, K_9   | 0:2, 1:8         | PX_X_DOWN & PX_X_UP
+Y       | K_4, K_0   | 0:3, 1:9         | PX_Y_DOWN & PX_Y_UP
+------------------------------------------------------
+PAUSE   | K_p        | 1:5              | PAUSE_DOWN & PAUSE_UP
+MENU    | K_m        | 0:4              | MENU_DOWN & MENU_UP
+"""
+import pygame.joystick
+from pygame import KEYDOWN, KEYUP, K_m, K_p, \
+    K_1, K_2, K_3, K_4, K_w, K_a, K_s, K_d, \
+    K_7, K_8, K_9, K_0, K_i, K_j, K_k, K_l, \
+    JOYBUTTONDOWN, JOYBUTTONUP, JOYAXISMOTION
+
+
+class InputManager():
+    def __init__(self):
+        joysticks = pygame.joystick.get_count()
+        if joysticks:
+            self.input_mode = 'JOY_CONTROL'
+        else:
```

Comment:
Lets break this PR into multiple smaller ones. This branch is called Marslander, so I'd expect it to have only mars lander related code, and maybe some small changes to other modules.

Also, there's like 50 images in this PR. I don't know how large a github repo can be, but were probably getting close to the size limit.